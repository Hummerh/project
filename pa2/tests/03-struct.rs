struct Point {x : i32, y : i32, z : i32}

struct Dog {
    Color: i32,
    Model: i32,
    Width: i32,
    Tailless: bool
}
