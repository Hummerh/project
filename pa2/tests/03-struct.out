(crate
    (items
        (struct-def
            (id
                (Point))
            (struct-field-defs
                (struct-field-def
                    (id
                        (x))
                    (type-i32))
                (struct-field-def
                    (id
                        (y))
                    (type-i32))
                (struct-field-def
                    (id
                        (z))
                    (type-i32))))
        (struct-def
            (id
                (Dog))
            (struct-field-defs
                (struct-field-def
                    (id
                        (Color))
                    (type-i32))
                (struct-field-def
                    (id
                        (Model))
                    (type-i32))
                (struct-field-def
                    (id
                        (Width))
                    (type-i32))
                (struct-field-def
                    (id
                        (Tailless))
                    (type-bool))))))